File: vipclaw-admin/src/main/java/com/vipamp/vipclaw/admin/service/AuthService.java
package com.vipamp.vipclaw.admin.service;

import com.vipamp.vipclaw.admin.dto.LoginRequest;
import com.vipamp.vipclaw.admin.dto.LoginResponse;

/**
 * 认证服务接口
 */
public interface AuthService {

    /**
     * 用户登录
     * @param request 登录请求
     * @return 登录响应
     */
    LoginResponse login(LoginRequest request);

    /**
     * 退出登录
     */
    void logout();
}

File: vipclaw-admin/src/main/kotlin/com/vipamp/vipclaw/admin/service/AuthService.kt
package com.vipamp.vipclaw.admin.service

import com.vipamp.vipclaw.admin.dto.LoginRequest
import com.vipamp.vipclaw.admin.dto.LoginResponse

/**
 * Authentication service interface
 */
interface AuthService {

    /**
     * User login
     * @param request Login request
     * @return Login response
     */
    fun login(request: LoginRequest): LoginResponse

    /**
     * Logout
     */
    fun logout()
}
